app/auth/annotations.py

```diff
@@ -33,23 +29,11 @@ def is_allowed(action):
     def wrapper(f):
         @wraps(f)
         def wrapped(*args, **kwargs):
-            entity_str, action_str = action.split("::")
             user_id, instance = None, None
             jwt_status, user_info = get_user_from_jwt(request, app.config['API_KEY'])
             if jwt_status:
                 user_id = user_info['user']
-
-            if entity_str == 'Package':
-                publisher_name, package_name = kwargs['publisher'], kwargs['package']
-                instance = Package.get_package(publisher_name, package_name)
-
-            elif entity_str == 'Publisher':
-                publisher_name = kwargs['publisher']
-                instance = Publisher.query.filter_by(name=publisher_name).one()
-            else:
-                return handle_error("INVALID_ENTITY", "{e} is not a valid one".format(e=entity_str), 401)
-
-            status = ia(user_id, instance, action)
+            status = get_status(action, kwargs['publisher'], kwargs['package'], user_id)
```

Comment:
Why the refactor of this into a separate file. A note on this in the commit message would have helped enormously.

Also `get_status` is a very undescriptive function name. As I understand it get_status is really "check_is_authorized" (??)